### ISSUE TITLE
Export does not work on Pixel running Android 10

### ISSUE THREAD

**Issue by GCrean** [beemdevelopment/Aegis#274]
I've just installed the latest from the PlayStore, Added 1 2FA account to try and export, It fails with the error 'an error occurred whilst trying to export the vault" I have given permission for the storage, So this is not the case.

Looking at the source, Would it be possible to change the export function to use Storage Access Framework ( https://developer.android.com/guide/topics/providers/document-provider.html )

This I believe could be the issue, Also would be a great feature to be able to export straight to google drive with no extra effort.

**Comment by michaelschattgen** [comment]
Thanks for your report! It seems like I can't reproduce this on my Pixel 2XL running Android 10. Exporting tokens works fine for me. The new storage access framework is definitely something we should look into but I don't think thats the problem here.

**Comment by alexbakker** [comment]
I also cannot reproduce this on my Pixel 3a XL running Android 10. Can you attach a debugger and see the exact exception that occurs?

Looking at the relevant piece of code, I noticed just now that we're using the deprecated (since API 29) ``getExternalFilesDir`` function. So this may actually have something to do with the way we access storage on the device, like @GCrean suggested. I don't really know much about the Storage Access Framework, but [``requestLegacyExternalStorage``](https://developer.android.com/reference/android/R.attr.html#requestLegacyExternalStorage) may be a workaround we can use while we look into a better solution.

**Comment by alexbakker** [comment]
This blog post has some good info: https://commonsware.com/blog/2019/06/07/death-external-storage-end-saga.html.

**Comment by GCrean** [comment]
Just loaded up in the debugger.
It doesn't give any logcat output, which is a bit annoying, However, the line below fails.

            if (!dir.exists() && !dir.mkdirs()) {
                throw new IOException("error creating external storage directory");
            }

The dir is "/storage/emulated/0/AegisDev"

I manually created the directory and it passed this test, The next issue was on the line 
   try (FileOutputStream stream = new FileOutputStream(file))

Giving the exception 

java.io.FileNotFoundException: /storage/emulated/0/AegisDev/aegis_export.json: open failed: EACCES (Permission denied)

Looks like a permissions issue to me,

As a quick test, I changed the line to
 File dir = new File(_context.getExternalFilesDir(null), dirName);

This worked. But puts it in a horrid place on the sdcard.

**Comment by GCrean** [comment]
btw, I tried android:requestLegacyExternalStorage="true" in the manifest and this didn't work :(

**Comment by alexbakker** [comment]
Thanks for your help debugging this issue.

So I initially tried to reproduce this on the production APK, but didn't succeed. After fiddling around with the debug APK a bit on my phone and an emulator, I encountered the same issue as you on both. The ``android:requestLegacyExternalStorage="true"`` workaround does work for me, though. Are you sure you placed it in the correct place in the manifest (i.e. under ``application``). Also, try reinstalling the APK after changing the manifest.

I then tried to reinstall the production APK and encountered the same issue. So now my real Aegis also can't export the vault, so that's great.
